hash.c

```diff
@@ -1895,12 +1895,18 @@ rb_check_hash_type(VALUE hash)
  *  call-seq:
  *    Hash.try_convert(obj) -> new_hash or nil
  *
+ *
  *  Returns the Hash object created by calling <tt>obj.to_hash</tt>:
  *    require 'csv' # => true
  *    row = CSV::Row.new(['Name', 'Age'], ['Bob', 45])
  *    row.respond_to?(:to_hash)  # => true
  *    Hash.try_convert(row) # => {"Name"=>"Bob", "Age"=>45}
  *
+ *  Returns the given <tt>obj</tt> if it is a Hash:
+ *    h = {}
+ *    h1 = Hash.try_convert(h)
+ *    h1.object_id == h.object_id # => true
```

Comment:
`equal?` would be better than `object_id` comparison.

Reply:
Don 't agree.  Not everyone will recognize that equal? means identity.  But object_id is highly intuitive.

Reply:
@BurdetteLamar  `#object_id` is not something we should encourage. If you write:

```
h1.equal?(h2) # identity comparison
```

maybe it's enough to be intuitive and informative too?